### PR TITLE
Make editor timeline zoom relative to song length

### LINKED ISSUE #19252: Waveform zooming-in increment depends on song length
### Type

Other

### Bug description

Songs shorter in length take less time to zoom in than longer songs.

eg, 34 second song can be set to max zoom in less than 150 scroll wheel inputs. Manual zooming is convenient and can be done in about 2 seconds.

https://user-images.githubusercontent.com/36340429/180041855-50040e9c-d523-4ca5-a041-129098d2a9fd.mp4

6 minute song takes more than 600 scroll wheel inputs to set to max zoom. Moreover, going to max zoom takes around 10 seconds with *scrolling automation*. Manual zooming would take much longer and is not a reasonable option.

https://user-images.githubusercontent.com/36340429/180042842-37c7b3b2-54d5-4af5-ad50-5d6ba1b0d112.mp4

Ideally, zooming in to max zoom would take as long as in the 34 second case for every song.

### Screenshots or videos

_No response_

### Version

2022.719.0-lazer

### Logs

[runtime.log](https://github.com/ppy/osu/files/9152314/runtime.log)
[updater.log](https://github.com/ppy/osu/files/9152315/updater.log)
[network.log](https://github.com/ppy/osu/files/9152316/network.log)
[performance.log](https://github.com/ppy/osu/files/9152317/performance.log)

### PR DESCRIPTION
**Pull request by Cwazywierdo**
Closes #19252

### CODE REVIEW

File: osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineArea.cs [hunk 1]
```diff
@@ -154,6 +156,6 @@ private void load(OverlayColourProvider colourProvider)
             Timeline.TicksVisible.BindTo(ticksCheckbox.Current);
         }
 
-        private void changeZoom(float change) => Timeline.Zoom += change;
+        private void changeZoom(float change) => Timeline.Zoom += change * Timeline.MaxZoom * zoom_button_sensitivity;
```
Comment (peppy) [inline osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineArea.cs:159]: I'd rather this be moved to a method inside `ZoomableScrollContainer` so the `const` / `MaxZoom` interaction doesn't need to be repeated here.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineArea.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineArea.cs:159]: I also don't think the math is very correct on this. If the intent was that the "step" of the zoom should be 0.02x (or 2%) of the allowable range, then this isn't accounting for `MinZoom` correctly. Should likely be something along the lines of `MinZoom + (MaxZoom - MinZoom) * zoom_button_sensitivity`.

**Comment by peppy** [comment]
@Cwazywierdo please check 4cec9a0, I've taken it a step further and combined both calls to actually perform the zoom (feels better to me).

**Review by peppy — APPROVED** [review]

**Comment by Cwazywierdo** [comment]
All looks good

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**